src/pipelinedefinition.cpp

```diff
@@ -1170,25 +1167,28 @@ shape_t PipelineDefinition::getNodeGatherShape(const NodeInfo& info) const {
             uint32_t demultiplyCount = someNodeInfo.demultiplyCount.value_or(0);
             if (demultiplyCount == 0) {
                 tensor_map_t nodeOutputsInfo;
-                auto result = PipelineDefinition::getCustomNodeMetadata(
-                    someNodeInfo,
-                    nodeOutputsInfo,
-                    someNodeInfo.library.getOutputsInfo,
-                    this->pipelineName);
-                if (!result.ok()) {
-                    SPDLOG_ERROR("Failed to read node: {} library metadata with error: {}", nodeName, result.string());
-                    return;
-                }
-                if (nodeOutputsInfo.size() == 0) {
-                    SPDLOG_ERROR("Node: {} library metadata reports no outputs", nodeName);
-                    return;
-                } else if (nodeOutputsInfo.begin()->second->getShape().size() < 3) {
-                    SPDLOG_ERROR("Node: {} library metadata reports output with too small number of dimensions", nodeName);
-                    return;
+                if (someNodeInfo.kind == NodeKind::CUSTOM) {
```

Comment:
I suppose models can also act as demultiplexer, do we handle it here?

Reply:
From the technical point there is no issue in having demultiplexer model however it was not considered in this part of code. I think from the reason that we block models that do have first dimension of input >1 from using in DAG with demultiplexer. Hence the only nodes that could be checked here is entry & custom node.

Reply:
What if model with batch=1 will be used as demultiplexer with demultiply_count=1?

Reply:
From my understanding we currently allow models to be demultiplexers. By not supporting this case here (ifs only for custom/entry node) we possibly introduce validation bugs.